engine/src/main/java/org/hibernate/validator/internal/xml/mapping/BeanStaxBuilder.java

```diff
@@ -174,25 +179,25 @@ void build(Set<Class<?>> processedClasses, Map<Class<?>, Set<ConstrainedElement>
 				constrainedElementsByType,
 				beanClass,
 				constrainedGetterStaxBuilders.stream()
-						.map( builder -> builder.build( beanClass, alreadyProcessedGetterNames ) )
+						.map( builder -> builder.build( javaBean, alreadyProcessedGetterNames ) )
 						.collect( Collectors.toList() )
 		);
 
-		List<Method> alreadyProcessedMethods = new ArrayList<>( constrainedMethodStaxBuilders.size() );
+		List<Callable> alreadyProcessedMethods = new ArrayList<>( constrainedMethodStaxBuilders.size() );
 		addConstrainedElements(
 				constrainedElementsByType,
 				beanClass,
 				constrainedMethodStaxBuilders.stream()
-						.map( builder -> builder.build( beanClass, alreadyProcessedMethods ) )
+						.map( builder -> builder.build( javaBean, alreadyProcessedMethods ) )
 						.collect( Collectors.toList() )
 		);
 
-		List<Constructor<?>> alreadyProcessedConstructors = new ArrayList<>( constrainedConstructorStaxBuilders.size() );
+		List<Callable> alreadyProcessedConstructors = new ArrayList<>( constrainedConstructorStaxBuilders.size() );
```

Comment:
I think we could talk about `JavaBeanConstructor` here? Or can't we?

Same above about using `JavaBeanMethod`.